From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Rochelemagne <rochelemagne@cpanel.net>
Date: Fri, 28 Sep 2018 11:43:07 -0500
Subject: [PATCH] Reduce startup size of PL_strtab

Perl 5.26 was using 512 as the default HvMAX for PL_strtab
this was increased in 528 to 2048, but we can start with
a lower value.

Note, that compiled programs using B::C would use one
accurate value from startup.

Changelog:
---
 SOURCES/perl/perl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SOURCES/perl/perl.c b/SOURCES/perl/perl.c
index 91bbbc60..f2ea2d50 100644
--- a/SOURCES/perl/perl.c
+++ b/SOURCES/perl/perl.c
@@ -353,7 +353,7 @@ perl_construct(pTHXx)
         /* SHAREKEYS tells us that the hash has its keys shared with PL_strtab,
          * which is not the case with PL_strtab itself */
         HvSHAREKEYS_off(PL_strtab);			/* mandatory */
-        hv_ksplit(PL_strtab, 1 << 11);
+        hv_ksplit(PL_strtab, 1 << 9); /* 5.26: 1 << 9 = 512 ; 5.28: 1 << 11 = 2048 */
     }
 
     Zero(PL_sv_consts, SV_CONSTS_COUNT, SV*);
